magento/magento2
# Use '.min' in filenames of already minified js files in the Swagger module so they aren't getting minified again in production, fixes #16927 - for Magento 2.2

**Please: do not forwardport this PR to 2.3, ~the solution for 2.3 is different and much cleaner, and~, it already exists: https://github.com/magento/magento2/pull/17627**

### Description
The Magento Swagger module contains already minified js files:
- https://github.com/magento/magento2/blob/2.2-develop/app/code/Magento/Swagger/view/frontend/web/swagger-ui/js/swagger-ui-bundle.js
- https://github.com/magento/magento2/blob/2.2-develop/app/code/Magento/Swagger/view/frontend/web/swagger-ui/js/swagger-ui-standalone-preset.js

~This PR marks them to not be minified again during static content deploy, because they get changed in a way that they are no longer valid JS.~
This PR renames them so they contain `.min` and reference them as such from the layout xml file as well. This way they don't get minified again in production mode when minification is enabled.

I've already gone ahead and created a fix for Magento 2.3 as well ~, since the solution is very different and much cleaner, this is due to https://github.com/magento/magento2/pull/13687 which only exists in 2.3~
PR for Magento 2.3: https://github.com/magento/magento2/pull/17627

### Fixed Issues (if relevant)
1. https://github.com/magento/magento2/issues/16927: 2.2.5 Swagger: With JS minification enabled, the swagger-ui-bundle.js becomes corrupted

### Manual testing scenarios
1. Set up a clean Magento 2.2.5 installation
2. Run: `bin/magento config:set dev/js/minify_files 1`
3. Run: `bin/magento deploy:mode:set production`
4. Go to: https://myinstallation.domain/swagger
5. See JS errors: `Uncaught SyntaxError: Invalid regular expression flag` & `Uncaught ReferenceError: SwaggerUIBundle is not defined`

### Contribution checklist
